src/Npgsql/Internal/TypeHandlers/MultirangeHandler.cs

```diff
@@ -9,7 +9,9 @@
 
 namespace Npgsql.Internal.TypeHandlers;
 
-public partial class MultirangeHandler<TSubtype> : NpgsqlTypeHandler<NpgsqlRange<TSubtype>[]>,
+// NOTE: This cannot inherit from NpgsqlTypeHandler<NpgsqlRange<TSubtype>[]>, since that triggers infinite generic recursion in Native AOT
+public partial class MultirangeHandler<TSubtype> : NpgsqlTypeHandler,
```

Comment:
Ah yeah, I think I did this for ranges in #4095, right?

Reply:
Yep, that's where I got the idea from.